kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/AclsController.java

```diff
@@ -26,20 +27,23 @@ public class AclsController extends AbstractController implements AclsApi {
 
   private final AclsService aclsService;
   private final AccessControlService accessControlService;
+  private final AuditService auditService;
 
   @Override
   public Mono<ResponseEntity<Void>> createAcl(String clusterName, Mono<KafkaAclDTO> kafkaAclDto,
                                               ServerWebExchange exchange) {
     AccessContext context = AccessContext.builder()
         .cluster(clusterName)
         .aclActions(AclAction.EDIT)
+        .auditOperation("createAcl")
         .build();
 
     return accessControlService.validateAccess(context)
-        .then(kafkaAclDto)
-        .map(ClusterMapper::toAclBinding)
-        .flatMap(binding -> aclsService.createAcl(getCluster(clusterName), binding))
-        .thenReturn(ResponseEntity.ok().build());
+        .<ResponseEntity<Void>>then(
+            kafkaAclDto.map(ClusterMapper::toAclBinding)
+                .flatMap(binding -> aclsService.createAcl(getCluster(clusterName), binding))
+                .thenReturn(ResponseEntity.ok().build())
+        ).doOnEach(sig -> auditService.audit(context, sig));
```

Comment:
**nit**: I don't mind but it feels like the "flat" chaining is doing the same. Isn't it?
```suggestion
        .then(kafkaAclDto)
        .map(ClusterMapper::toAclBinding)
        .flatMap(binding -> aclsService.createAcl(getCluster(clusterName), binding))
        .doOnEach(sig -> auditService.audit(context, sig))
        .thenReturn(ResponseEntity.ok().build());
```

Reply:
@AndreyNenashev good catch, it is a legacy from first refactoring